src/composer.ts

```diff
@@ -16,19 +16,23 @@ export default abstract class implements Composable {
 
   protected _contracts!: Contracts;
 
-  public web3: Web3;
+  public blockchainApi: Web3;
 
   /**
    * Create RNS library.
    *
    * @remarks
-   * If web3 points to RSK Mainnet or RSK Testnet, no options are required. Contract addresses are detected automatically.
+   * If the blockchain api points to RSK Mainnet or RSK Testnet, no options are required. Contract addresses are detected automatically.
    *
-   * @param web3 - Web3 instance
+   * @param blockchainApi - Web3 or Rsk3 instance
    * @param options - Overrides network defaults. Optional on RSK Mainnet and RSK Testnet, required for other networks.
    */
-  constructor(web3Instance: Web3, options?: Options) {
-    this.web3 = new Web3(web3Instance.currentProvider);
+  constructor(blockchainApi: Web3 | any, options?: Options) {
```

Comment:
A thought - and definitely something to experiment with in a future PR, not this one - use the type definitions that you have for web3.js for rsk3.js that should surface any differences in the API.